fakecamera/fakecamera.go

```diff
@@ -0,0 +1,393 @@
+
+    camera_specs := map[string]interface{}{
+        headers.YResolution: camera.ResY(),
+        headers.XResolution: camera.ResX(),
+        headers.FrameSize:   lepton3.BytesPerFrame,
+        headers.Model:       lepton3.Model,
+        headers.Brand:       lepton3.Brand,
+        headers.FPS:         camera.FPS(),
+    }
+
+    cameraYAML, err := yaml.Marshal(camera_specs)
+    if _, err := conn.Write(cameraYAML); err != nil {
+        return err
+    }
+
+    conn.Write([]byte("\n"))
+    log.Printf("Listening for send frames ...")
+    return queueLoop(conn)
+}
+
+func Send(params url.Values) {
+    enq, err := strconv.ParseBool(params.Get("enqueue"))
+    if err != nil {
+        enq = false
+    }
+    if !enq {
+        ClearQueue(true)
+    }
+    queue.enqueue(params)
+}
+
+func queueLoop(conn *net.UnixConn) error {
+    for {
+        stopSending = false
+        params := queue.dequeue()
+        if params == nil {
+            time.Sleep(queueSleep)
+            continue
+        }
+        repeat, _ := strconv.Atoi(params.Get("repeat"))
+        if repeat == 0 {
+            repeat = 1
+        }
+
+        generate, err := strconv.ParseBool(params.Get("generate"))
+        if err != nil {
+            generate = false
+        }
+        if generate {
+            sendFrames(conn, params, repeat)
+        } else {
+            for i := 0; i < repeat; i++ {
+                err := sendCPTV(conn, params)
+                if err != nil {
+                    return err
+                }
+            }
+        }
+    }
+}
+
+func forceStop() {
+    stopSending = true
+}
+
+func ClearQueue(stop bool) {
+    queue.clear()
+    if stop {
+        forceStop()
+    }
+}
+
+func makeFrame(minTemp, maxTemp int, hotspots []hotspot) *cptvframe.Frame {
+    var pix int
+    out := cptvframe.NewFrame(camera)
+    for y, row := range out.Pix {
+        for x, _ := range row {
+            if maxTemp <= minTemp {
+                pix = minTemp
+            } else {
+                pix = minTemp + rand.Intn(maxTemp-minTemp)
+            }
+            out.Pix[y][x] = uint16(pix)
+        }
+    }
+    addHotpsots(out.Pix, hotspots)
+    return out
+}
+
+func addHotpsots(pix [][]uint16, hotspots []hotspot) {
+    if hotspots == nil {
+        return
+    }
+
+    for _, hotspot := range hotspots {
+        hotspot.addSpot(pix)
+    }
+}
+
+func setStatus(telemetry *cptvframe.Telemetry, timeon time.Duration, ffc bool, plusMS int) {
+    telemetry.TimeOn = timeon + time.Duration(plusMS)*time.Millisecond
+    if ffc {
+        telemetry.FFCState = "FFCRunning"
+        telemetry.LastFFCTime = timeon + time.Duration(plusMS)*time.Millisecond
+    }
+}
+func sendCPTV(conn *net.UnixConn, params url.Values) error {
+    file := params.Get("cptv-file")
+    fullpath := path.Join(cptvDir, file)
+    if _, err := os.Stat(fullpath); err != nil {
+        log.Printf("%v does not exist\n", fullpath)
```

Comment:
Does it need too? the logs will show in the screen running the camera.  Otherwise the we can store the last error and add an endpoint to retrieve the error